src/mcm_chapter_9/stack/Stack.java

```
@@ -0,0 +1,60 @@
+package mcm_chapter_9.stack;
+
+
+//A stack class for characters.
+public class Stack {
+	
+	private char stck[]; // this array holds the stack
+	
+	private int tos; // top of stack
+	
+	//Construct an empty Stack given its size.
+	public Stack(int size) {
+		stck = new char[size]; // allocate memory for stack
+		tos = 0;
+	}
+	
+	//Construct a Stack from a Stack.
+	public Stack( Stack ob ) {
+		tos = ob.tos;
+		stck = new char[ob.stck.length];
+	
+		//copy elements
+		for(int i=0; i < tos; i++){
+			stck[i] = ob.stck[i];
+		}
+	}
+	
+	//Construct a stack with initial values.
+	public Stack(char a[]) {
+		stck = new char[a.length];
+		for(int i = 0; i < a.length; i++) {
+			try {
+				push( a[i] );
+			}
+			catch ( StackFullException e ) {
```

Would probably be better to just throw the exception, no?